lib/animina_web/live/profile_live.ex

```diff
@@ -45,8 +45,22 @@ defmodule AniminaWeb.ProfileLive do
           {current_user_green_flags, current_user_red_flags} =
             fetch_green_and_red_flags(current_user.id, language)
 
+          intersecting_green_flags_count =
```

Comment:
I don't like how we hit the database multiple times for flag related stuff. We fetch :white, :red, :green and again :red and :green. That is just waisting computing power for nothing. Please refactor and rename `fetch_green_and_red_flags/2` to do one database request for all colours of that given user and split it within Elixir and do all the needed calculations.

Reply:
One could even argue that this all should be done in the resource (e.g. with `aggregates`).

@MICHAELMUNAVU83 Fire at will or don't we can kick this refactoring can down the road.

Reply:
But I don't want too many database hits. Keep it to a minimum.